### DISCUSSION TITLE
How extract cookies for subsequent connection test

### DISCUSSION THREAD

**Discussion by nicholaswkc34** [locustio/locust#2841] [General]
Dear forumers, I have a login request which the response assign to response variable as global scope. How to extract the token or something from the request??

**Comment by cyberw** [comment]
Hi!

You can access the [CookieJar](https://pydoc.dev/requests/latest/requests.cookies.RequestsCookieJar.html) for the Session at `self.client.cookies`. FastHttpUser has a slightly different structure, available at `self.client.client.cookiejar`.

**Comment by nicholaswkc34** [comment]
I had follow your advise to use the cookies in login request but how to use the cookie in subsequent after login request. 

Below is my code: 

```
global response
    global myCookies
    
    @task
    def homepage(self):
        self.client.get("/")

    @task
    def getEvent(self):
        self.client.get("event/tik-tok-award-2024-realisasikan-impianmu")

    @task
    def checkout(self):
        self.client.get("event/checkout")

    @task
    def login(self):
        response = self.client.post("https://www,test.org",
                         {"username":"0", "password":"0!" })
        myCookies = self.client.cookies
```

Please help as i new to python.

**Comment by cyberw** [comment]
Ah. To reuse cookies as a normal browser does, you shouldnt need to do anything at all, I thought you were trying to do something more advanced. 

Check the response from the POST that does the login to confirm that it actually worked and contains the headers to set the cookie.

This might be helpful: https://docs.locust.io/en/stable/running-in-debugger.html#print-http-communication

**Comment by nicholaswkc34** [comment]
I want to use the cookies in the following request(Check Account and order history function). How to do it?

Below is my code: 

```
import logging
from locust import HttpUser, task, between
from http.client import HTTPConnection

class WebsiteUser(HttpUser):
    wait_time = between(5, 10)

    HTTPConnection.debuglevel = 1
    logging.basicConfig()
    logging.getLogger().setLevel(logging.DEBUG)
    requests_log = logging.getLogger("requests.packages.urllib3")
    requests_log.setLevel(logging.DEBUG)
    requests_log.propagate = True

    global response
    global myCookies

    @task
    def homepage(self):
        self.client.get("/")

    @task
    def getEvent(self):
        self.client.get("event/tik-tok-award-2024-realisasikan-impianmu")

    @task
    def checkout(self):
        self.client.get("event/checkout")

    @task
    def login(self):
        response = self.client.post("https://stg-accounts.mediaprimaplus.com.my/authorize?client_id=632354224783531&redirect_uri=https%3A%2F%2Fstaging-ticket.revid.my%2Fcallback&response_type=code&state=%2F",
                         {"username":"0102200614", "password":"Cheeliangan15!" })
        myCookies = self.client.cookies


    #@task
    #def checkAccount(self):
    #    self.client.get("account?tab=account")

    #@task
    #def orderHistory(self):
    #    self.client.get("account?tab=order")
```

The check account and order history function need login before can fire the request. How to use the cookies in these function?

**Comment by cyberw** [comment]
Like I said, you don't need to do anything at all to persist cookies. 

However, in your locustfile, there is nothing to ensure login happens _before_ your other requests. To fix that, either put all your requests in a single task with the login-POST first, or define an on_start method where you do the login, see documentation.

If that doesn't help, you need check the debug log from actually running the test and examine the response from the login request.
